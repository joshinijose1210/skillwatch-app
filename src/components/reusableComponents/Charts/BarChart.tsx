import { RatingChartData } from '@constants/data';
import { ResponsivePieCanvas } from '@nivo/pie';
import { useState } from 'react';
import { ChartTitle, ChartWrapper } from './Chart.styled';
import { BarChartProps } from './types';

export const BarChart = ({ rating }: BarChartProps) => {
    const ratingData = RatingChartData[Math.floor(rating) - 1];
    ratingData.value = (360 / 5) * rating;
    const initialColor = [
        {
            id: '',
            label: 'initial',
            value: 360 - ratingData.value,
            color: '#cfd4dc'
        }
    ];
    const [chartData] = useState([ratingData, initialColor[0]]);
    return (
        <ChartWrapper>
            <ChartTitle>
                {rating}
                <br />
                <span>Average</span>
            </ChartTitle>
            <ResponsivePieCanvas
                data={rating > 0 ? chartData : initialColor}
                colors={rating > 0 ? chartData.map(({ color }) => color) : initialColor.map(({ color }) => color)}
                enableArcLabels={false}
                enableArcLinkLabels={false}
                margin={{ top: 30, right: 30, bottom: 30, left: 20 }}
                innerRadius={0.75}
                padAngle={0}
                endAngle={360}
                cornerRadius={0}
                activeOuterRadiusOffset={8}
                borderWidth={1}
                borderColor={{
                    from: 'color',
                    modifiers: [['darker', 0.2]]
                }}
                arcLinkLabelsSkipAngle={10}
                arcLinkLabelsTextColor="#333333"
                arcLinkLabelsThickness={2}
                arcLinkLabelsColor={{ from: 'color' }}
                arcLabelsSkipAngle={45}
                arcLabelsTextColor={{
                    from: 'color',
                    modifiers: [['darker', 2]]
                }}
            />
        </ChartWrapper>
    );
};
