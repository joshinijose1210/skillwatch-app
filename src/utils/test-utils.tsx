import { ToastContainer } from '@medly-components/core';
import { setupListeners } from '@reduxjs/toolkit/dist/query';
import { store as rtkStore, setupStore } from '@slice';
import { RenderOptions, RenderResult, render } from '@testing-library/react';
import { defaultTheme } from '@theme';
import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import reduxMockStore from 'redux-mock-store';
import { ThemeProvider } from 'styled-components';
import { userMockState } from '../mocks/userMockState';
import { initialState } from '../slice';

interface CustomOptions extends RenderOptions {
    initialState?: any;
    initialEntries?: any;
}

interface WithThemeProviderProps {
    children?: React.ReactNode;
}

const mockStore = reduxMockStore(),
    store = mockStore({
        ...initialState,
        user: { ...initialState.user, groups: ['admin'] }
    });

const WithThemeProvider: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <ThemeProvider theme={defaultTheme}>
        <>
            <ToastContainer position="top-end" />
            {children}
        </>
    </ThemeProvider>
);

const WithStore: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <Provider store={store}>
        <WithThemeProvider>{children}</WithThemeProvider>
    </Provider>
);

const WithRouter: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <MemoryRouter>
        <WithThemeProvider>{children}</WithThemeProvider>
    </MemoryRouter>
);

const WithStoreAndRouter: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <Provider store={store}>
        <MemoryRouter>
            <WithThemeProvider>{children}</WithThemeProvider>
        </MemoryRouter>
    </Provider>
);

export const RTKRender = (ui: React.ReactElement, { initialState = userMockState, ...renderOptions }: CustomOptions = {}) => {
    const store = setupStore(initialState);
    setupListeners(store.dispatch);

    const RtkWrapper: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
        <Provider store={store}>
            <MemoryRouter initialEntries={renderOptions.initialEntries}>
                <WithThemeProvider>{children}</WithThemeProvider>
            </MemoryRouter>
        </Provider>
    );
    return { rtkStore, ...render(ui, { wrapper: RtkWrapper, ...renderOptions }) };
};
const customRender = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithThemeProvider, ...options });

export const renderWithStore = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithStore, ...options });

export const renderWithRouter = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithRouter, ...options });

export const renderWithStoreAndRouter = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithStoreAndRouter, ...options });

export const wrapper = (props: { children: React.ReactNode }, { initialState = userMockState }: CustomOptions = {}) => {
    const store = setupStore(initialState);
    setupListeners(store.dispatch);
    return (
        <Provider store={store}>
            <MemoryRouter>
                <WithThemeProvider>{props.children}</WithThemeProvider>
            </MemoryRouter>
        </Provider>
    );
};

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render, store };
