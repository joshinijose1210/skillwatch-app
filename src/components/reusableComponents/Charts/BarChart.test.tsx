import { renderWithStoreAndRouter } from '@utils/test-utils';
import { BarChart } from './BarChart';

describe('bar chart', () => {
    it('should render properly', () => {
        const { container } = renderWithStoreAndRouter(<BarChart rating={1} />);
        expect(container).toMatchSnapshot();
    });
});
