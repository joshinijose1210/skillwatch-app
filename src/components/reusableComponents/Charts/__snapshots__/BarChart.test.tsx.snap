// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bar chart should render properly 1`] = `
<div>
  <div
    class="sc-qgqgvp-7 wJwVI"
  >
    <div
      class="sc-qgqgvp-8 cZRUgi"
    >
      1
      <br />
      <span>
        Average
      </span>
    </div>
    <div
      style="width: 100%; height: 100%;"
    />
  </div>
</div>
`;
